src/claviska/SimpleImage.php

```diff
@@ -118,58 +161,36 @@ public function fromDataUri($uri) {
    * Loads an image from a file.
    *
    * @param string $file The image file to load.
+   * @param boolean $sslVerify Set to false to skip SSL validation.
    * @throws \Exception Thrown if file or image data is invalid.
    * @return \claviska\SimpleImage
    */
   public function fromFile($file) {
-    // Check if the file exists and is readable. We're using fopen() instead of file_exists()
-    // because not all URL wrappers support the latter.
-    $handle = @fopen($file, 'r');
-    if($handle === false) {
+    // Set fopen options.
+    $sslVerify = $this->getFlag("sslVerify"); // Don't perform peer validation when true
+    $opts = [
+      "ssl" => [
+        "verify_peer"      => $sslVerify,
+        "verify_peer_name" => $sslVerify
+      ]
+    ];
+
+    // Check if the file exists and is readable.
+    $file = @file_get_contents($file, false, stream_context_create($opts));
+    if($file === false) {
       throw new \Exception("File not found: $file", self::ERR_FILE_NOT_FOUND);
     }
-    fclose($handle);
+
+    // Create image object from string
+    $this->image = imagecreatefromstring($file);
 
     // Get image info
-    $info = @getimagesize($file);
+    $info = @getimagesizefromstring($file);
     if($info === false) {
       throw new \Exception("Invalid image file: $file", self::ERR_INVALID_IMAGE);
     }
     $this->mimeType = $info['mime'];
 
-    // Create image object from file
-    switch($this->mimeType) {
-    case 'image/gif':
-      // Load the gif
-      $gif = imagecreatefromgif($file);
-      if($gif) {
-        // Copy the gif over to a true color image to preserve its transparency. This is a
-        // workaround to prevent imagepalettetruecolor() from borking transparency.
-        $width = imagesx($gif);
-        $height = imagesy($gif);
-        $this->image = imagecreatetruecolor((int) $width, (int) $height);
-        $transparentColor = imagecolorallocatealpha($this->image, 0, 0, 0, 127);
-        imagecolortransparent($this->image, $transparentColor);
-        imagefill($this->image, 0, 0, $transparentColor);
-        imagecopy($this->image, $gif, 0, 0, 0, 0, $width, $height);
-        imagedestroy($gif);
-      }
```

Comment:
I'm mostly good with this, but I'm concerned that we'll break GIF transparency by using `imagecreatefromstring()` instead. We probably shouldn't discard this code.

Reply:
> I'm concerned that we'll break GIF transparency by using `imagecreatefromstring()`

I tested this in my original PR and was unable to reproduce a situation where the alpha channel was lost or broken with a handful of transparent GIFs and `imagecreatefromstring()` in GD 2.3.0. 

Strangely, I can't find any mention of this being patched in GD but I can find some older posts about this being an issue, so I agree with you, let's keep it in. Will push a fix ASAP.